### ISSUE TITLE
Can't use externalized syntax highlighting

### ISSUE THREAD

**Issue by 00sapo** [onivim/oni#2552]
**Oni Version:** 0.3.6
**Neovim Version (Linux only):** 0.3.1
**Operating System:** Manjaro 17.1.12, Linux 4.14.66-1-MANJARO x86_64

**Issue:** Can't use externalized syntax highlighting

**Expected behavior:** When I want to use a syntax highlighting provided by some Vim plugin and I disable TextMate syntax, the one coming from the plugin should be used

**Actual behavior:** Default syntax is shown

**Steps to reproduce:**

Use SemanticHighlighting plugin:

```
syntax on
Plug 'jaxbot/semantic-highlight.vim'
```

Disable TextMate:
```
"editor.textMateHighlighting.enabled": false,
"oni.loadInitVim": true,
```

**Comment by CrossR** [comment]
I think this may be due to how the plugin checks if a GUI is running.
The plugin checks the following to check if it should use GUI colours:

https://github.com/jaxbot/semantic-highlight.vim/blob/01e646ca17dfe0143ef9c8a04a684e8f100bd20d/plugin/semhl.vim#L144

Of which none are set in Oni, due to either not making sense to enable or not existing to enable.

If I manually force the plugin to work in Oni (by adding an `exists('g:gui_oni')` to the check above), then the plugin works as expected:

<img width="428" alt="after" src="https://user-images.githubusercontent.com/10038688/45089700-eed93d00-b104-11e8-90fe-bfc8787af7f3.PNG">

However, that is obviously not a fix at all.
I'm not sure what the general nvim way of doing this is, except for checking `nvim_list_uis()` and checking the RGB flag it returns (which Oni does set). Perhaps someone more versed in how to do this could jump in and say?

But after a change is made to the plugin, the setup you've got should work as expected. Its maybe a different issue of if Oni should stop the textmate highlights if it detects some new user enabled ones.

**Comment by 00sapo** [comment]
You are saying that the best way is to make SemanticHighlight compatible to oni, right? I think that since there are two plugins that don't work because of oni, maybe oni itself could try to make an effort about this issue... But I'm not a Vim developer, so maybe I'm wrong.

**Comment by CrossR** [comment]
>You are saying that the best way is to make SemanticHighlight compatible to oni, right?

Not quite, more just to be compatible with any Neovim GUI, not just Oni. Ie, the current approach means it also doesn't work in nvim-qt, NyaoVim and I assume any other neovim gui that uses GUI colours.

There is a PR that would fix it for some of them (https://github.com/jaxbot/semantic-highlight.vim/pull/61) but it sounds like setting `&termiguicolors` in Oni isn't really appropriate, as per @justinmk's comment on the Goyo issue.

Using `nvim_list_uis()` seems to be the way to work for all nvim GUIs from now on, not just Oni. I think since the nvim GUIs (including Oni!) are still mainly fairly early on, some plugins don't react to them very well yet, and also why not many plugins have adapted to them.

**Comment by 00sapo** [comment]
Ok, for now I have added support to Oni in my fork version of SemanticHighlight. I plan to come back and implement the `nvim_list_uis()` way.

**Comment by badosu** [comment]
Shouldn't we make `gui_running` be true on Oni?

**Comment by CrossR** [comment]
>Shouldn't we make gui_running be true on Oni?

I don't know if that exists in the nvim side of things. I did a `:helpgrep gui_running` and the only reference I found to it was in the ALE config. If it does, then possibly, but my very basic understanding is that it relates to GVim rather than a GUI in general?

**Comment by badosu** [comment]
>  If it does, then possibly, but my very basic understanding is that it relates to GVim rather than a GUI in general?

Yes, also Macvim.

However it seems that this is the case because they were the only GUIs available at the time. I think it would be sensible to use this as many plugins may use this variable to detect capabilities, e.g. instead of using `t_Co` detecting if it is a GUI, etc..

**Comment by badosu** [comment]
@justinmk Thanks for the feedback!
